### PR TITLE
Support private registries

### LINKED ISSUE #164: Get registry to use from config.toml
`cargo-update` currently has the index url hard-coded, but sometimes users will have a separate index configured at `~/.cargo/config.toml`

### PR DESCRIPTION
**Pull request by Logarithmus**
Support private registries
    
See https://github.com/nabijaczleweli/cargo-update/issues/164#issuecomment-965558321
    
Closes #164

### CODE REVIEW

**Comment by nabijaczleweli** [comment]
Maybe the first commit, sure, but drop the one where you rewrite the entire repository for no reason lol, and the second one that breaks 1.35.0 builds, for seemingly also no reason?

**Comment by Logarithmus** [comment]
@nabijaczleweli why do you insist on using 2.5 years old version of Rust? Just remove this job from CI.
I "rewrite the entire repository" for a reason, and the reason is to make your `rustfmt.toml` work, since you didn't update it in 3 years, and some options have changed since then.

**Comment by nabijaczleweli** [comment]
(a) MSRV, I don't see a reason to break that interface, and (b) excuse you, I *do* use rustfmt and this is the config for it. If your version doesn't like it, then the answer is to leave it unformatted, not rewrite the repository, destroying any and all hope of git blame being useful again.

**Comment by Logarithmus** [comment]
@nabijaczleweli I've removed clippy & rustfmt commits, as you wish. But still don't get why anyone want to use 2.5 years old Rust. As you can see, there are no public crates using your project as a dependency: https://lib.rs/crates/cargo-update/rev
Also you can just release `9.0.0` & break any interfaces, those who use `8.0.0` won't suffer any breakage at all.

P.S. What does MSRV mean?

File: src/ops/mod.rs [hunk 1]
```diff
@@ -1100,24 +1101,48 @@ pub fn get_index_url(crates_file: &Path, registry: &str) -> Result<(String, Cow<
...
+                )
+            })
+            .filter(|(_, url)| url.is_some())
+            .map(|(name, url)| (name, url.unwrap()))
+            .for_each(|(name, url)| {
+                if cur_source == **url {
+                    cur_source = name.into()
+                }
+                registries.insert(name, url.into());
+            })
```
Comment (nabijaczleweli) [inline src/ops/mod.rs:1144]: what a mess, how is this different from
```rust
        for (name, url) in table.iter()
            .flat_map(|(name, val)| val.as_table().and_then(|t| t.get("index")).and_then(|idx| idx.as_str()).map(|v| (name, v))) {
            if cur_source == url {
                cur_source = name.into()
            }
            registries.insert(name, url.into());
        }
```
or, indeed,
```rust
        for (name, url) in table.iter().flat_map(|(name, val)| val.as_table()?.get("index")?.as_str().map(|v| (name, v))) {
            if cur_source == url {
                cur_source = name.into()
            }
            registries.insert(name, url.into());
        }
```
?

**Review by nabijaczleweli — CHANGES_REQUESTED** [review]
In all of these, you decided to ignore values with a wrong type instead of erroring on them, like it was so far. Why?

File: src/ops/mod.rs (hunk 1 above)
Comment (Logarithmus) [reply src/ops/mod.rs]: Your solution is pretty neat. I've updated this PR.

**Comment by nabijaczleweli** [comment]
Looks good!

**Comment by nabijaczleweli** [comment]
Released in [`v8.1.0`](https://github.com/nabijaczleweli/cargo-update/releases/tag/v8.1.0)!

**Pull request merged**